src/components/DSixInput/index.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+
+const DSixInput = ({ className }) => {
+  const [value, setValue] = useState("");
+
+  const sanitizeInput = (rawInput) => {
+    const currentValue = value;
+    const validValueWhitelist = ["", "1", "2", "3", "4", "5", "6"];
+
+    if (rawInput.length < 0 || rawInput.length > 2) {
+      return currentValue;
+    }
+
+    const newValue = rawInput.replace(currentValue, "");
```

Comment:
I'm not entirely sure what this accounts for

Reply:
`rawInput` is possibly a 2-digit number where one digit is equal to `currentValue`. This removes that digit and just leaves the newly typed one, that way you can update the number in the input without having to hit backspace.

Reply:
I'm not sure what you mean by "selecting the input on focus". I thought I explained the use case in my last comment. 🤔 I'm probably just misunderstand what you're asking about here. I'll answer your larger question in the other comment and maybe that will clarify things.